iOS/MSMapsModules/MSMapsModules/MSMapGeoJsonParser/MSMapGeoJsonParser.m

```diff
@@ -251,42 +318,133 @@ + (MSGeoposition * _Nullable)parseGeoposition:(NSArray *)coordinates
   } else {
 
     *error = [MSMapGeoJsonParser
-        makeGeoJsonError:[@"coordinates array must contain "
-                          @"at least latitude and longitude. Instead saw: "
-                             stringByAppendingString:
-                                 [coordinates componentsJoinedByString:@", "]]];
+        makeGeoJsonError:
+            [@"position array must contain "
+             @"at least latitude and longitude. Instead saw: "
+                stringByAppendingString:[MSMapGeoJsonParser
+                                            arrayToString:coordinates]]];
     return nil;
   }
 }
 
-+ (MSGeopath *)parsePath:(NSArray *)pathArray error:(NSError **)error {
-  NSArray<MSGeoposition *> *path = [[NSArray alloc] init];
-  MSMapAltitudeReferenceSystem altitudeReferenceSystem =
-      MSMapAltitudeReferenceSystemSurface;
-  for (id obj in pathArray) {
-    if (![obj isKindOfClass:[NSArray class]]) {
-      *error = [MSMapGeoJsonParser
-          makeGeoJsonError:
-              [@"coordinates of LineString must contain an array of "
-               @"positions. Instead saw: "
-                  stringByAppendingFormat:@"%@", (NSString *)obj]];
-      return nil;
-    }
++ (MSGeopath *)parsePath:(NSArray *)pathArray
+    altitudeReferenceSystem:
+        (MSMapAltitudeReferenceSystem)altitudeReferenceSystem
+                      error:(NSError **)error {
+  if (pathArray.count == 0) {
+    *error = [MSMapGeoJsonParser
+        makeGeoJsonError:
+            [@"Geopath must contain an array of positions. Instead saw: "
+                stringByAppendingString:[MSMapGeoJsonParser
+                                            arrayToString:pathArray]]];
+    return nil;
+  }
 
+  NSMutableArray<MSGeoposition *> *path = [[NSMutableArray alloc] init];
+  for (id obj in pathArray) {
     NSArray *latLong = (NSArray *)obj;
-    MSGeoposition *position = [self parseGeoposition:latLong error:error];
+    bool useAltitude = false;
+    if (altitudeReferenceSystem == MSMapAltitudeReferenceSystemEllipsoid) {
```

Comment:
Consider shortening this to:
bool useAltitude = altitudeReferenceSystem == MSMapAltitudeReferenceSystemEllipsoid;